qualtran/bloqs/cswap_lth_reg.py

```diff
@@ -0,0 +1,122 @@
+    iteration length, $n_b$ is the bitsize of
+    the registers to swap, and $n_c$ is the number of controls.
+
+    Args:
+        selection_regs: Indexing `select` signature of type Tuple[`SelectionRegisters`, ...].
+            It also contains information about the iteration length of each selection register.
+        bitsize: The size of the registers we want to swap.
+        nth_gate: A function mapping the composite selection index to a single-qubit gate.
+
+    Registers:
+        control_registers: Control registers
+        selection_regs: Indexing `select` signature of type Tuple[`SelectionRegisters`, ...].
+            It also contains information about the iteration length of each selection register.
+        target_registers: Target registers to swap. We swap FROM registers
+            labelled x`i`, where i is an integer and TO a single register called y
+
+    References:
+        [Fault-Tolerant Quantum Simulations of Chemistry in First Quantization](
+            https://arxiv.org/abs/2105.12767) page 20 paragraph 2.
+    """
+    selection_regs: Tuple[SelectionRegister, ...] = attr.field(
+        converter=lambda v: (v,) if isinstance(v, SelectionRegister) else tuple(v)
+    )
+    bitsize: int
+    control_regs: Tuple[Register, ...] = attr.field(
+        converter=lambda v: (v,) if isinstance(v, Register) else tuple(v),
+        default=(Register('ctrl', 0),),
+    )
+
+    @cached_property
+    def control_registers(self) -> Tuple[Register, ...]:
+        return self.control_regs
+
+    @cached_property
+    def selection_registers(self) -> Tuple[SelectionRegister, ...]:
+        return self.selection_regs
+
+    @cached_property
+    def target_registers(self) -> Tuple[Register, ...]:
+        iteration_length = self.selection_registers[0].iteration_length
+        regs = [Register(f'x{i}', bitsize=self.bitsize) for i in range(iteration_length)]
+        regs += [Register('y', bitsize=self.bitsize)]
+        return tuple(regs)
+
+    def _circuit_diagram_info_(self, args: cirq.CircuitDiagramInfoArgs) -> cirq.CircuitDiagramInfo:
+        wire_symbols = ["@"] * total_bits(self.control_registers)
+        wire_symbols += ["In"] * total_bits(self.selection_registers)
+        for i, target in enumerate(self.target_registers):
+            if i == len(self.target_registers) - 1:
+                wire_symbols += [f"×(y)"] * target.total_bits()
+            else:
+                wire_symbols += [f"×(x)"] * target.total_bits()
+        return cirq.CircuitDiagramInfo(wire_symbols=wire_symbols)
+
+    def nth_operation(
+        self, context: cirq.DecompositionContext, control: cirq.Qid, **kwargs
+    ) -> cirq.OP_TREE:
+        selection_idx = tuple(kwargs[reg.name] for reg in self.selection_registers)[0]
+        target_regs = {reg.name: kwargs[reg.name] for reg in self.target_registers}
+        return CSwap(self.bitsize).make_on(
+            ctrl=control, x=target_regs[f'x{selection_idx}'], y=target_regs['y']
+        )
```

Comment:
Again, support the more general case using 
```suggestion
        selection_idx = tuple(kwargs[reg.name] for reg in self.selection_registers)
        target_regs = kwargs['targets']
        return CSwap(self.bitsize).make_on(
            ctrl=control, x=target_regs[selection_idx], y=target_regs['y']
        )
```

Reply:
Done.